### PR TITLE
Add search endpoint

### PR DESCRIPTION
**Pull request by pettermk**

### CODE REVIEW

File: tag_hierarchy/commands/search.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
+//
+// Created by Petter Moe Kvalvaag on 2019-10-04.
```
Comment (erikmaland) [inline tag_hierarchy/commands/search.cpp:2]: I guess this date is left from another file?

File: tag_hierarchy/commands/search.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
+// Created by Petter Moe Kvalvaag on 2019-10-06.
```
Comment (erikmaland) [inline tag_hierarchy/commands/search.h:2]: Same here with date.

**Review by erikmaland — APPROVED** [review]
Nice!

File: tag_hierarchy/commands/search.cpp (hunk 1 above)
Comment (pettermk) [reply tag_hierarchy/commands/search.cpp:2]: I think that is actually when this was started, better late than never 😆 
On the topic, IDEs tend to leave these things but I'd rather delete them, agree?

Comment (erikmaland) [reply tag_hierarchy/commands/search.cpp:2]: Aha! I am fine with either way :)

**Pull request merged**